include/zephyr/sys/libc-hooks.h

```diff
@@ -41,13 +41,12 @@ __syscall size_t zephyr_fwrite(const void *ZRESTRICT ptr, size_t size,
 
 /* Handle deprecated malloc arena size configuration values */
 #ifdef CONFIG_COMMON_LIBC_MALLOC
-# if CONFIG_COMMON_LIBC_MALLOC_ARENA_SIZE == 0
+# if defined(CONFIG_MINIMAL_LIBC) && (CONFIG_MINIMAL_LIBC_MALLOC_ARENA_SIZE != -2)
 #  undef CONFIG_COMMON_LIBC_MALLOC_ARENA_SIZE
-#  ifdef CONFIG_MINIMAL_LIBC
-#   define CONFIG_COMMON_LIBC_MALLOC_ARENA_SIZE CONFIG_MINIMAL_LIBC_MALLOC_ARENA_SIZE
-#  else
-#   define CONFIG_COMMON_LIBC_MALLOC_ARENA_SIZE 0
-#  endif
+#  define CONFIG_COMMON_LIBC_MALLOC_ARENA_SIZE CONFIG_MINIMAL_LIBC_MALLOC_ARENA_SIZE
+# elif defined(CONFIG_PICOLIBC) && (CONFIG_PICOLIBC_HEAP_SIZE != -2)
```

Comment:
> With this change, the old Kconfig options now take priority over the new ones

Since setting these deprecated Kconfigs do not print a deprecation warning (because they are `int` and cannot select `DEPRECATED`), I think it would help if we print a warning if these deprecated Kconfigs are set to a non-default value.

Something like:

```
# if defined(CONFIG_MINIMAL_LIBC) && (CONFIG_MINIMAL_LIBC_MALLOC_ARENA_SIZE != -2)
#    warning ...
...
```

Reply:
yeah, that's a good idea -- will also trigger compile errors when built with -Werror